stacks-signer/src/signer.rs

```diff
@@ -1453,3 +1506,116 @@ impl Signer {
         Ok(())
     }
 }
+
+fn load_encrypted_signer_state<S: SignerStateStorage>(
+    storage: S,
+    id: S::IdType,
+    private_key: &Scalar,
+) -> Result<Option<v2::Signer>, PersistenceError> {
+    if let Some(encrypted_state) = storage.get_encrypted_signer_state(id)? {
+        let serialized_state = decrypt(private_key, &encrypted_state)?;
+        let state = serde_json::from_slice(&serialized_state)
+            .expect("Failed to deserialize decryoted state");
+        Ok(Some(v2::Signer::load(&state)))
+    } else {
+        Ok(None)
+    }
+}
+
+trait SignerStateStorage {
+    type IdType;
+
+    fn get_encrypted_signer_state(
+        self,
+        signer_config: Self::IdType,
+    ) -> Result<Option<Vec<u8>>, PersistenceError>;
+}
```

Comment:
I like the way you designed and use this trait 👍

Reply:
Thanks :blush:

Reply:
Seconded.